assets/msalv2.js

```diff
@@ -12,35 +12,44 @@ var aadOauth = (function () {
   };
 
   // Initialise the myMSALObj for the given client, authority and scope
-  function init(config) {
-    // TODO: Add support for other MSAL configuration
-    var msalConfig = {
-      auth: {
-        clientId: config.clientId,
-        authority: config.isB2C ? "https://" + config.tenant + ".b2clogin.com/" + config.tenant + ".onmicrosoft.com/" + config.policy + "/" : "https://login.microsoftonline.com/" + config.tenant,
-        redirectUri: config.redirectUri,
-      },
-      cache: {
-        cacheLocation: "localStorage",
-        storeAuthStateInCookie: false,
-      },
-    };
-
-    if (typeof config.scope === "string") {
-      tokenRequest.scopes = config.scope.split(" ");
-    } else {
-      tokenRequest.scopes = config.scope;
-    }
-
-    tokenRequest.extraQueryParameters = JSON.parse(config.customParameters);
-    tokenRequest.prompt = config.prompt;
-
-    myMSALObj = new msal.PublicClientApplication(msalConfig);
-    // Register Callbacks for Redirect flow and record the task so we
-    // can await its completion in the login API
-
-    redirectHandlerTask = myMSALObj.handleRedirectPromise();
-  }
+ function init(config) {
+     // TODO: Add support for other MSAL configuration
+     var authData = {
+         clientId: config.clientId,
+         authority: config.isB2C ? "https://" + config.tenant + ".b2clogin.com/" + config.tenant + ".onmicrosoft.com/" + config.policy + "/" : "https://login.microsoftonline.com/" + config.tenant,
+         redirectUri: config.redirectUri,
+     };
+     var postLog = {
+         postLogoutRedirectUri: config?.postLogoutRedirectUri,
```

Comment:
Thank you for the changes @Musaddiq625. One last thing, `config` can not be null or undefined at this point, could you remove the `?` before the point.